client/src/components/Instructions/index.js

```diff
@@ -270,6 +364,77 @@ export default class Instructions extends Component {
     );
   }
 
+    renderGaslessUpgrade() {
+      return (
+        <div className={styles.instructions}>
+          <h2> Upgrading your contract with access control </h2>
+          <p> Thanks to ZeppelinOS, you can upgrade the code of your contract to add more functionality. With this functionality we can restrict access to relayed transactions. </p>
+          <div className={styles.step}>
+            <div className={styles.instruction}>
+              1. Open <span>contracts/GaslessCounter.sol</span> and change line 45
+            </div>
+            from:
+            <div className={styles.code}>
+              <code>
+                {`require(!whiteList[from]);`}
+              </code>
+            </div>
+             to:
+           <div className={styles.code}>
+              <code>
+                {`require(whiteList[from]);`}
+              </code>
+            </div>
+          </div>
+          <div className={styles.step}>
+            <div className={styles.instruction}>
+              2. Save the changes and compile and push the new changes to the network.
+            </div>
+            <div className={styles.code}>
+              <code>
+                zos push
+              </code>
+            </div>
+          </div>
+          <div className={styles.step}>
+            <div className={styles.instruction}>
+              3. Update the already deployed contract with the new code
+            </div>
+            <div className={styles.code}>
+              <code>
+                zos update GaslessCounter
+              </code>
+            </div>
+          </div>
+          <div className={styles.step}>
+            <div className={styles.instruction}>
+              4. Open up the truffle console
+            </div>
+            <div className={styles.code}>
+              <code>
+                truffle console
+              </code>
+            </div>
+          </div>
+          <div className={styles.step}>
+            <div className={styles.instruction}>
+              5. In truffle console, add your metamask address to the whitelist. (NOTE: Replace {`<YOUR-METAMASK-ADDRESS>`} with your metamask address)
```

Comment:
It should be possible to include here YOUR-METAMASK-ADDRESS directly, so we already show the user the command they need to copy-paste, instead of having them replace it manually. I think it's already implemented in the regular Counter tutorial.

Reply:
Done , adding to next commit